e2e/cypress/integration/channel/channel_menu_spec.js

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+import users from '../../fixtures/users.json';
+
+function demoteUserToGuest(user) {
+    // # Issue a Request to demote the user to guest
+    const baseUrl = Cypress.config('baseUrl');
+    cy.externalRequest({user: users.sysadmin, method: 'post', baseUrl, path: `users/${user.id}/demote`});
+}
+
+function promoteGuestToUser(user) {
+    // # Issue a Request to promote the guest to user
+    const baseUrl = Cypress.config('baseUrl');
+    cy.externalRequest({user: users.sysadmin, method: 'post', baseUrl, path: `users/${user.id}/promote`});
+}
+
+describe('Channel header menu', () => {
+    before(() => {
+        cy.loginAsNewUser().as('newuser');
```

Comment:
This will fail if guest account is not enable in the config.
Before logging in as new user, please add lines to update config, something like:
```javascript
cy.apiUpdateConfig({
    GuestAccountsSettings: {
        Enable: true,
    },
});
```

Reply:
Good point, I'll do it, and I'm thinking now that I can't cherry-pick that to 5.15 because there is not guest accounts, so probably the simplest approach is to cherry-pick the code and not cherry-pick the cypress test.

Reply:
Done